### PR TITLE
bpo-30463 ABC has empty __slots__, with unittest

### PR DESCRIPTION
**Pull request by aaronchall**

### CODE REVIEW

**Review by ilevkivskyi — APPROVED** [review]
I think this is a simple but good idea.

**Comment by aaronchall** [comment]
Would be cool to have ABC available in 2.7...

So what happens next, should I or someone else click "Close and comment"?

**Comment by ilevkivskyi** [comment]
> Would be cool to have ABC available in 2.7...

It is not possible at this stage.

> So what happens next, should I or someone else click "Close and comment"?

This PR can be approved and merged by someone with commit rights.
Maybe @serhiy-storchaka has time for this?

**Comment by serhiy-storchaka** [comment]
LGTM. Please add a Misc/NEWS entry. Don't forget "Patch by Aaron Hall." at the end.

File: Misc/NEWS [hunk 1]
```diff
@@ -1028,10 +1028,14 @@ Library
...
   classes to use keyword parameters in __init_subclass__. Patch by Nate Soares.
 
+- Issue #30463: Add ``__slots__ = ()`` to ``abc.ABC`` to allow
+  subclassers to deny ``__dict__`` and ``__weakref__`` creation.
+  Patch by Aaron Hall.
+
 - Issue #25532: inspect.unwrap() will now only try to unwrap an object
   sys.getrecursionlimit() times, to protect against objects which create a new
   object on every attribute access.
-
```
Comment (ilevkivskyi) [inline Misc/NEWS]: Could you please remove the extra whitespace you added?

**Comment by aaronchall** [comment]
trailing whitespace removed and tests passed

**Review by serhiy-storchaka — APPROVED** [review]
This doesn't much matter, but usually Misc/NEWS entries are added at the start of the section.

In any case the PR LGTM.

**Pull request merged**